SplitStringToArray/Program.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+
+namespace SplitStringToArray
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            string cities = "London, Paris, Rom";
+
+            var citiesArray = SplitStringToArray(cities, ',');
+
+            foreach (var city in citiesArray)
+            {
+                Console.WriteLine(city);
+            }
+
+            Console.ReadLine();
+        }
+
+        static string[] SplitStringToArray(string input, char charSplit)
+        {
+            input = input.Replace(" ", "");
+            var countSplitSymbol = input.Where(x => x == charSplit).Count();
+
+            string[] result = new string[countSplitSymbol+1];
+
+            var charArray = input.ToCharArray();
+
+            var builder = new StringBuilder();
+
+            int j = 0;
+
+            for (int i = 0; i < charArray.Length; i++)
+            {
+                if (charArray[i] == charSplit)
+                {
+                    j++;
+                    builder = new StringBuilder();
+                    continue;
+                }
+                else
+                {
+                    result[j] = builder.Append(charArray[i]).ToString();
```

Comment:
тобто при кожнії ітерації створюється нова строка и перезаписується стара? це дуже не ефіктивно.

Reply:
тут мені трішки не зручно заперечувати, але метод ToString викликається тут тільки один раз при роботі стрінг білдера із змінною, наприклад поки змінна j = 0 стрінг білдер буде наповнювати її а викликатись метод ToString лиш один раз. В даному прикладі метод ToString буде викликатись 3 рази. А також запитав чат джпт, відповідь нижчеЖ

hello how many times will work method ToString() in C# in this method  => public static string[] SplitStringToArray(string input)
{
    input = input.Replace(" ", "");
    var countSplitSymbol = input.Where(x => x == ',').Count();

    string[] result = new string[countSplitSymbol+1];

    var charArray = input.ToCharArray();

    var builder = new StringBuilder();

    int j = 0;

    for (int i = 0; i < charArray.Length; i++)
    {
        if (charArray[i] == ',')
        {
            j++;
            builder = new StringBuilder();
            continue;
        }
        else
        {
            result[j] = builder.Append(charArray[i]).ToString();
        }
    }

    return result;
}
if input will be like this "Лондон, Париж, Рим";

Let's analyze how many times the `ToString()` method will be called in your `SplitStringToArray` method when the input is `"Лондон, Париж, Рим"`.

Initial Input: The input string is "Лондон, Париж, Рим".

Removing Spaces: The input.Replace(" ", "") will change the input to "Лондон,Париж,Рим".

Counting Split Symbols: The line var countSplitSymbol = input.Where(x => x == ',').Count(); will count the number of commas in the string:

There are 2 commas, so countSplitSymbol will be 2.



Creating Result Array: The result array will be initialized as string[] result = new string[countSplitSymbol + 1];, which means it will have a size of 2 + 1 = 3.

Converting String to Char Array: The string is converted to a character array, which will be charArray = { 'Л', 'ондон', ',', 'П', 'ариж', ',', 'Р', 'им' }.

Iterating through Characters:

For each character in charArray, the following happens:

When the character is a comma (,), the index j is incremented, and a new StringBuilder is created. This means the builder is reset.

For other characters, they are appended to the current StringBuilder, and ToString() is called to convert the StringBuilder to a string.





Characters Processed:

Characters processed:

Л → appended (ToString called)

о → appended

н → appended

д → appended

о → appended

н → appended → At this point, ToString() is called when the next character (,) appears.

, → resets the builder, increments j.

П → appended (ToString called)

а → appended

р → appended

и → appended

ж → appended → At this point, ToString() is called when the next character (,) appears.

, → resets the builder, increments j.

Р → appended (ToString called)

и → appended

м → appended → At this point, ToString() is NOT called because the loop ends.






Total Calls to ToString()

The `ToString()` method will be called:

Once when we encounter 'Л' (first substring: "Лондон").

Once when we encounter 'П' (second substring: "Париж").

Once when we encounter 'Р' (third substring: "Рим").


Conclusion

In total, `ToString()` will be called **3 times** for the input `"Лондон, Париж, Рим"`.

Reply:
А ти спробуй подебижити.
В тебе умова стоіть що коли зустрічається деліметер, тоді робити новий стриг билдер, а в іншому випадку додавати чар до текщого стрингбілдера, створювати нову строку та класти її на місце старої.

щоб працювало так як ти пишешь терба зробити щось таке
```
for (int i = 0; i < charArray.Length; i++)
{
    if (charArray[i] == charSplit)
    {
        result[j] = builder.ToString();
        j++;
        builder.Clear();
        continue;
    }
    else
    {
        builder.Append(charArray[i]);
    }
}
```